### PR TITLE
Closes #11 vs_metabolic test data 

### LINKED ISSUE #11: Test data: `vs_metabolic`
### Background Information

We need to create an `{admiralmetabolic}` version of `pharmaversesdtm::vs` because the patients in the existing `vs` dataset are not obese.

This dataset will eventually be moved to `{pharmaversesdtm}`, but for the moment we will create it within `{admiralmetabolic}`.

### Definition of Done

- [ ] top-level `data_raw` folder created to house `vs_metabolic`, i.e. the script that generates our metabolic version of `vs`
- [ ] `vs_metabolic` program set up. program should read in `vs`, keep only a few patients and edit all relevant tests (weight, BMI, etc) to make them obese.
- [ ] New `vs.metabolic` dataset documented in `data.R`

### PR DESCRIPTION
**Pull request by Siddhesh2097**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@Siddhesh2097 thanks, looks much better now - will review as soon as i get a chance :)

File: data-raw/vs_metabolic.R [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+vs1_hipcir <- vs1 %>%
+  filter(VSTESTCD == "WEIGHT") %>%
+  mutate(VSTESTCD = "HIPCIR", VSTEST = "Hip Circumference")
+
+# Bind new parameter records to original dataset
+vs_full <- bind_rows(vs1, vs1_wstcir, vs1_hipcir) %>%
+  arrange(USUBJID, VSTESTCD, VISITNUM, VSDY)
+
+# Updating each parameters values for each subjects to make them obese
```
Comment (manciniedoardo) [inline data-raw/vs_metabolic.R:37]:
```suggestion
# Updating each parameter values for each subject to make them obese
```

File: data-raw/vs_metabolic.R [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+library(dplyr)
+library(pharmaversesdtm)
+
+## Read input data ----
+
+data("vs")
+data("dm_metabolic")
+
+# Convert blank to NA ----
+vs <- convert_blanks_to_na(vs)
```
Comment (manciniedoardo) [inline data-raw/vs_metabolic.R:16]: Do we need this line? Isn't `pharmaversesdtm::vs` already in the right format? 

In any case`convert_blanks_to_na()` comes from `{admiral}` and so you'd need to add a `library(admiral)` call at the top of your script. But because this code will eventually be moved to `{pharmaversesdtm}`, which `{admiral}` depends on, this would create a sort of circular dependency.

As such I'd remove or find another way to do this step.

File: data-raw/vs_metabolic.R [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    VSSTRESU = VSORRESU,
+  ) %>%
+  arrange(USUBJID, VSTESTCD, VISITNUM, VSDY)
+# get common column names
+
+common_cols <- intersect(names(vs_mb), names(vs))
+
+# assign labels to mb_vs
+
+walk(common_cols, \(x) attr(vs_mb[[x]], "label") <<- attr(vs[[x]], "label"))
```
Comment (manciniedoardo) [inline data-raw/vs_metabolic.R:283]: You need to add `library(purrr)` at the top as `walk()` is exported from `{purrr}`

File: data-raw/vs_metabolic.R [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
```
Comment (manciniedoardo) [inline data-raw/vs_metabolic.R:284]: Can you reorder the `vs_metabolic` columns so that they are in the same order as for `vs`?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @Siddhesh2097 this is super. Just a couple of minor comment and then we can merge :)

File: data-raw/vs_metabolic.R (hunk 1 above)
Comment (Siddhesh2097) [reply data-raw/vs_metabolic.R:37]: Updated

File: data-raw/vs_metabolic.R (hunk 3 above)
Comment (Siddhesh2097) [reply data-raw/vs_metabolic.R:283]: Agreed. Removed it

File: data-raw/vs_metabolic.R (hunk 2 above)
Comment (Siddhesh2097) [reply data-raw/vs_metabolic.R:16]: Agreed. Removed it

**Comment by Siddhesh2097** [comment]
Hi @manciniedoardo , Updated

**Review by manciniedoardo — APPROVED** [review]
